server/data/data-api/src/test/java/org/apache/james/rrt/lib/AliasResolverContract.java

```diff
@@ -0,0 +1,167 @@
+/** *************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ***************************************************************/
+
+package org.apache.james.rrt.lib;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.Username;
+import org.apache.james.rrt.api.AliasResolver;
+import org.junit.jupiter.api.Test;
+
+import com.github.fge.lambdas.Throwing;
+
+public interface AliasResolverContract {
+
+    Domain DOMAIN = Domain.of("example.com");
+    Domain OTHER_DOMAIN = Domain.of("other.org");
+    Username USER = Username.of("user@example.com");
+    Username USER_ALIAS = Username.of("alias@example.com");
+    Username OTHER_USER = Username.of("other@example.com");
```

Comment:
Do we handle users without domain (eg: virtualHosting = false)

Reply:
No, but that's the aliases' behavior

Reply:
Anyway, I bet we need to revise RRT behaviour in the first hand when virtual hosting is turned off...

Reply:
Can we do it in another PR

Reply:
Yes for sure.